File: pe003.py
#code created by NamanNimmo Gera
#10:50am, April 11, 2019.

def largestPrime(num):
    j = 2
    while num % j != 0:
        j += 1
    if num % 2 == 0:
        if num == 2:
            return 2
        else:
            return largestPrime(num/2)
    elif num / j == 1:
        return num
    else:
        return largestPrime(num / j)


x = 600851475143
print(int(largestPrime(x)))

File: pe003.go
package main

import "fmt"

func LargestPrimeFactor(num int) int {
	j := 2

	for (num % j) != 0 {
		j += 1
	}

	if (num % 2) == 0 {
		if num == 2 {
			return 2
		} else {
			return LargestPrimeFactor(num / 2)
		}
	} else if num/j == 1 {
		return num
	} else {
		return LargestPrimeFactor(num / j)
	}
}

func main() {
   fmt.Println(LargestPrimeFactor(600851475143)) // 6857
}
